src/main/nodejs/src/app/user-profile/user-profile.component.ts

```diff
@@ -47,4 +50,14 @@ export class UserProfileComponent implements OnInit {
     });
   }
 
+  deleteAccount(modalContent) {
+    this.modalService.open(modalContent).result.then(() => {
+      this.accountService.deleteUser().subscribe(() => {
+        this.accountService.logout().subscribe(() => {
+          this.accountService.refreshCache();
+          this.router.navigate(['/login'], { queryParams: { delete: ''}});
+        });
```

Comment:
The `subscribe()` call will not longer be necessary if the `logout()` method submits a form instead.